### PR TITLE
fix double sign request 

### PR DESCRIPTION
**Pull request by AlexeyTuboltsev**
this is a small followup for #1837 
fix OnlyPublicRoute to only redirect on mount

### CODE REVIEW

File: app/components/shared/routing/OnlyPublicRoute.tsx [hunk 1]
```diff
@@ -34,9 +29,9 @@ export const OnlyPublicRoute = compose<IComponentProps, RouteProps>(
       isAuthorized: selectIsAuthorized(s.auth),
     }),
   }),
-  branch<IStateProps>(
-    state => state.isAuthorized,
-    renderComponent(OnlyPublicRouteRedirectComponent),
-  ),
+  onEnterAction({
+    actionCreator: (dispatch, props) =>
+      props.isAuthorized && dispatch(actions.routing.goToDashboard()),
```
Comment (desfero) [inline app/components/shared/routing/OnlyPublicRoute.tsx]: IMO when we want to do some side-effects it's better to use `if`. with `{ }` as this then easier to read. What do you think?

File: app/components/shared/routing/OnlyPublicRoute.tsx [hunk 2]
```diff
@@ -34,9 +29,9 @@ export const OnlyPublicRoute = compose<IComponentProps, RouteProps>(
       isAuthorized: selectIsAuthorized(s.auth),
     }),
   }),
-  branch<IStateProps>(
-    state => state.isAuthorized,
-    renderComponent(OnlyPublicRouteRedirectComponent),
-  ),
+  onEnterAction({
+    actionCreator: (dispatch, props) =>
```
Comment (desfero) [inline app/components/shared/routing/OnlyPublicRoute.tsx]: can we type `props` as now it's `any`?

File: app/components/shared/routing/OnlyPublicRoute.tsx [hunk 3]
```diff
@@ -15,12 +16,6 @@ interface IComponentProps {
   component: React.ComponentType;
 }
 
-type TProps = RouteProps & IStateProps;
-
-const OnlyPublicRouteRedirectComponent: React.FunctionComponent<TProps> = props => {
```
Comment (desfero) [inline app/components/shared/routing/OnlyPublicRoute.tsx]: What was the exact problem here? Seems like logic is still the same?

Comment (AlexeyTuboltsev) [reply app/components/shared/routing/OnlyPublicRoute.tsx]: yeah I thought it is but it's not. We should only redirect on component mount and now it redirects on props change too. That's what I fixed with this pr

File: app/components/shared/routing/OnlyPublicRoute.tsx (hunk 1 above)
Comment (AlexeyTuboltsev) [reply app/components/shared/routing/OnlyPublicRoute.tsx]: I think you're right 😄

File: app/components/shared/routing/OnlyPublicRoute.tsx (hunk 3 above)
Comment (drakkein) [reply app/components/shared/routing/OnlyPublicRoute.tsx]: yeah, looks like you are right @AlexeyTuboltsev

**Pull request merged**